MyNumSetLibrary/MyNumSetLibrary/NumSet.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Collections.Generic;
+
+namespace MyNumSetLibrary
+{
+    public class NumSet
+    {
+        private readonly HashSet<int> _HashSet;
+
+        public NumSet(params int[] hashSet)
+        {
+            _HashSet = new HashSet<int>(hashSet);
+        }
+
+        public override string ToString()
+        {
+            return string.Join(",", _HashSet);
+        }
+
+        public override bool Equals(object? obj)
+        {
+            if (ReferenceEquals(this, obj))
+            {
+                return true;
+            }
+
+            if (obj is not NumSet)
+            {
+                return false;
+            }
+
+            return _HashSet.SetEquals(((NumSet)obj)._HashSet);
+        }
+
+        public override int GetHashCode()
+        {
+            int hashCode = 0;
+
+            foreach (int i in _HashSet)
+            {
+                hashCode += i;
+            }
+
+            return hashCode + _HashSet.Count;
+        }
+
+        public static bool operator ==(NumSet left, NumSet right)
+        {
+            if (left is null && right is null)
+            {
+                return true;
+            }
+
+            if (left is null ^ right is null)
+            {
+                return false;
+            }
+
+            return left!.Equals(right);
+        }
+
+        public static bool operator !=(NumSet left, NumSet right)
+        {
+            return !(left == right);
+        }
+
+        public int[] ReturnArray()
+        {
+            int[] array = new int[_HashSet.Count];
+            _HashSet.CopyTo(array);
+
+            return array;
```

Comment:
This works, but there is a shortcut with LINQ.
```suggestion
            return _HashSet.ToArray();
```